src/views/Dashboard.vue

```diff
@@ -246,8 +246,10 @@ export default {
     )
     this.subscribe('root')
   },
-  beforeDestroy () {
+  beforeRouteLeave (to, from, next) {
+    this.unsubscribe('root')
```

Comment:
Surely this should be done by workflowSevice.unregister?

Reply:
Ah, all this time I thought I needed both. Looks like the correct is

- Component/view wants to use the service:
  * `register()`
  * `subscribe()`
- Component is being destroyed or doesn't need the data anymore (e.g. a lumino widget removed)
  * `unregister` (which will do the same as `unsubscribe()`)

@oliver-sanders do we need the `unsubscribe methods`? Both in `GQuery.js`, and in all the views? Could we perhaps combine register/subscription in a single method? 

Updating the PR to use only `unregister`.

Reply:
Done. I removed one unsubscribe call but left only some unsubscribe calls in `Workflow.vue` view. The way I built it, when you create that view, it "registers" the query. Then each time you create a widget it is "subscribed". Then when you remove the widget it just "unsubscribe"s. Then, when you leave the page it now only "unregisters".

Changing that here would increase considerably the amount of changes in this PR. But we can review it later (the goal here is just making sure we have the right amount of subscriptions when you navigate)